sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java

```diff
@@ -126,8 +133,8 @@ public ResourceBuilder setValue(Value value) {
      * Assigns a unique resource ID for this resource, which is used to uniquely identify it in later offer evaluation
      * runs. This may be used with e.g. restarting a task at its prior location.
      */
-    public ResourceBuilder setResourceId(Optional<String> resourceId) {
-        this.resourceId = resourceId;
+    public ResourceBuilder setResourceId(String resourceId) {
+        this.resourceId = Optional.of(resourceId);
```

Comment:
Would we expect `resourceId` to ever be `null`? Should this be `Optional.ofNullable`?